packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,632 @@
+      // Otherwise, it's a single date or time, so only check `start`
+      return props.range || hasTimePeriods.value
+        ? !selectedRange.value.start || !selectedRange.value.end
+        : !selectedRange.value.start
+    })
+    const defaultTimeframe = computed(() => {
+      return props.defaultRelative !== undefined
+        ? props.defaultRelative
+        : null
+    })
+
+    /**
+     * Updates our internal (read: separate) state of currently selected `v-calendar` value(s)
+     * @param {object | string} vCalValue Object containing a pair of `start` and `end` timestamps,
+     * or a single timestamp.
+     */
+    const changeCalendarRange = (vCalValue) => {
+      let start = ''
+      let end = ''
+
+      // If value is an object, this is a time range. Else, a single date or time value.
+      if (vCalValue.hasOwnProperty('start')) {
+        start = vCalValue.start
+        end = vCalValue.end
+      } else {
+        start = vCalValue
+      }
+
+      // Set emitted value when v-calendar selection is made.
+      // The `timePeriodsKey` param only applies to relative timeframes,
+      // not `v-calendar` selections; however, this keeps the object "shape" consistent.
+      selectedRange.value = {
+        start: start ? new Date(start) : '',
+        end: end ? new Date(end) : '',
+        timePeriodsKey: ''
+      }
+    }
+
+    /**
+     * Updates both the input field value, and the full time frame readout
+     * when a relative time frame button is clicked
+     * @param {*} timeframe
+     */
+    const changeRelativeTimeframe = (timeframe) => {
+      selectedTimeframe.value = timeframe
+
+      // Format the start/end values as human readable date
+      const start = selectedTimeframe.value.start()
+      const end = selectedTimeframe.value.end()
+
+      // Set value to be emitted when relative time frame clicked
+      selectedRange.value = {
+        start: new Date(start),
+        end: new Date(end),
+        timePeriodsKey: selectedTimeframe.value.key
+      }
+
+      fullRangeDisplay.value = formatDisplayDate({ start, end })
+    }
+
+    /**
+     * Clears any previously made choices, and emits the result of this action
+     * back to the parent.
+     */
+    const clearSelection = () => {
+      selectedCalendarRange.value = null
+      abbreviatedDisplay.value = props.placeholder
+      fullRangeDisplay.value = ''
+      selectedRange.value = { start: '', end: '', timePeriodsKey: '' }
+
+      if (hasTimePeriods) {
+        selectedTimeframe.value = props.timePeriods[0]
+      }
+
+      // If the calendar has focus, and `range` is set to false, we are displaying
+      // a single date / time picker; therefore, need to emit an empty string.
+      //
+      // Else, we are displaying a date / time range (relative timeframes or calendar range)
+      if (!props.range && showCalendar) {
+        emit('changed', '')
+      } else {
+        emit('changed', selectedRange.value)
```

Comment:
We should emit `change` (present tense) not `changed`


_Similar to a `click` event is not a `clicked` event_


You'll need to update all occurrences and usage. 

```suggestion
        emit('change', '')
      } else {
        emit('change', selectedRange.value)
```